src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/SuggestionHandlers/StrInterpSuggestionHandler.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Syntax;
+using Microsoft.PowerFx.Core.Syntax.Nodes;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Texl.Intellisense
+{
+    internal partial class Intellisense
+    {
+        internal sealed class StrInterpSuggestionHandler : NodeKindSuggestionHandler
+        {
+            public StrInterpSuggestionHandler()
+                : base(NodeKind.StrInterp)
+            {
+            }
+
+            internal override bool TryAddSuggestionsForNodeKind(IntellisenseData.IntellisenseData intellisenseData)
+            {
+                Contracts.AssertValue(intellisenseData);
+
+                var curNode = intellisenseData.CurNode;
+                var cursorPos = intellisenseData.CursorPos;
+
+                var strInterpNode = curNode.AsStrInterp();
+                var spanMin = strInterpNode.Token.Span.Min;
+
+                if (cursorPos < spanMin)
+                {
+                    // Cursor is before the head
+                    // i.e. | $"...."
+                    // Suggest possibilities that can result in a value.
+                    IntellisenseHelper.AddSuggestionsForValuePossibilities(intellisenseData, strInterpNode);
+                }
+                else if (strInterpNode.Token.Span.Lim > cursorPos || strInterpNode.StrInterpEnd == null)
+                {
+                    // Handling the erroneous case when user enters a space after $ and cursor is after space.
+                    // Cursor is before the open quote.
+                    // Eg: $ | "
+                    return false;
+                }
+                else
```

Comment:
Does this need to handle `$"{ | ` or would that be an Error node instead of StringInterp?

Reply:
I added tests for this